aspnetcore/blazor/security/content-security-policy.md

```diff
@@ -243,67 +305,74 @@ Add additional `script-src` and `style-src` hashes as required by the app. Durin
 
 The particular script associated with the error is displayed in the console next to the error.
 
-### Server-side Blazor apps
+:::moniker range=">= aspnetcore-8.0"
 
-In the [`<head>` content](xref:blazor/project-structure#location-of-head-and-body-content), apply the directives described in the *Policy directives* section:
+## Apply a CSP in non-Development environments
 
-:::moniker range=">= aspnetcore-6.0"
+When a CSP is applied to a Blazor app's `<head>` content, it interferes with local testing in the Development environment. For example, [Browser Link](xref:client-side/using-browserlink) and the browser refresh script fail to load. The following examples demonstrate how to apply the CSP's `<meta>` tag in non-Development environments.
 
-```html
-<meta http-equiv="Content-Security-Policy" 
-      content="base-uri 'self';
-               default-src 'self';
-               img-src data: https:;
-               object-src 'none';
-               script-src 'self';
-               style-src 'self';
-               upgrade-insecure-requests;">
+> [!NOTE]
+> The examples in this section don't show the full `<meta>` tag for the CSPs. The complete `<meta>` tags are found in the subsections of the [Apply the policy](#apply-the-policy) section earlier in this article.
+
+Three general approaches are available:
+
+* Apply the CSP via the `App` component, which applies the CSP to all layouts of the app.
+* Apply the CSP to the app's layout files using the [`<HeadContent>` tag](xref:blazor/components/control-head-content). For complete effectiveness, every app layout file must adopt the approach.
```

Comment:
Perhaps, custom combos of external scripts would be permitted on a per-layout basis. I felt that since it's a valid approach and might be useful that we should list it ... as you suggest tho, not in the first position.